tools/scaffold/scaffold.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+const (
+	ResourceCmd         = "resource"
+	DataSourceCmd       = "data-source"
+	PluralDataSourceCmd = "plural-data-source"
+)
+
+// struct which is applied to go template files
+type ScaffoldParams struct {
+	GenerationType    string
+	NamePascalCase    string
+	NameCamelCase     string
+	NameSnakeCase     string
+	NameLowerNoSpaces string
+}
+
+type FileGeneration struct {
+	TemplatePath string
+	OutputPath   string
+}
+
+func main() {
+	nameCamelCase := os.Args[1]
+	generationType := os.Args[2]
+
+	params := ScaffoldParams{
+		GenerationType:    generationType,
+		NamePascalCase:    ToPascalCase(nameCamelCase),
+		NameCamelCase:     nameCamelCase,
+		NameSnakeCase:     ToSnakeCase(nameCamelCase),
+		NameLowerNoSpaces: strings.ToLower(nameCamelCase),
+	}
+
+	files, err := filesToGenerate(&params)
+	if err != nil {
+		panic(err)
+	}
+	for _, file := range files {
+		if err := generateFileFromTemplate(file, &params); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func filesToGenerate(params *ScaffoldParams) ([]FileGeneration, error) {
+	folderPath := fmt.Sprintf("internal/service/%s", params.NameLowerNoSpaces)
+
+	switch params.GenerationType {
+	case ResourceCmd:
+		return []FileGeneration{
+			{
+				TemplatePath: "tools/scaffold/template/resource.tmpl",
```

Comment:
I think at the moment is great to run the tool with go run and having the templates in a folder.
in the future if we want to create a binary for the tool we may want to use "embed" to have all template files in the binary